src/box/alter.cc

```diff
@@ -1089,15 +1101,58 @@ class CheckSpaceFormat: public AlterSpaceOp
 	virtual void prepare(struct alter_space *alter);
 };
 
+/**
+ * The object is used to grant ability to yield with RAII approach.
+ * Transaction is allowed to yield only on its first statement, so if the
+ * statement is not first, it simply does nothing.
+ * If it's the first statement, the guard blocks execution until all previous
+ * alters will be rolled back or committed so that the space object won't be
+ * deleted right from under our feet. In the case when the previous alters were
+ * rolled back and the space was removed from space cache, the constructor
+ * throws an error.
+ */
+class AlterYieldGuard
+{
+public:
+	AlterYieldGuard(struct space *old_space) {
+		if (!txn_is_first_statement(in_txn()))
+			return;
+		txn_can_yield(in_txn(), true);
+		uint32_t space_id = old_space->def->id;
+		while (true) {
+			bool space_is_being_altered = false;
+			struct alter_space *alter;
+			rlist_foreach_entry(alter, &alter_space_list, in_list) {
+				if (alter->txn != in_txn() &&
+				    alter->old_space->def->id == space_id) {
+					space_is_being_altered = true;
+					break;
+				}
+			}
+			if (!space_is_being_altered)
+				break;
+			fiber_sleep(0.0);
```

Comment:
Looks good but please don't busy-loop. Let's wait on `fiber_cond` instead (may be global).

Reply:
I thought about `fiber_cond` per `alter`, but it looked like overkill. Global one is a good idea, thanks.
Done.